src/parser.js

```diff
@@ -143,6 +145,54 @@ export default class Parser extends Stream {
                 });
                 return;
               }
+
+              // check if the content is encrypted for Widevine
+              if (entry.attributes.KEYFORMAT === widevineUuid) {
+                const VALID_METHODS = ['SAMPLE-AES', 'SAMPLE-AES-CTR', 'SAMPLE-AES-CENC'];
+
+                if (VALID_METHODS.indexOf(entry.attributes.METHOD) === -1) {
+                  this.trigger('warn', {
+                    message: 'invalid key method provided for Widevine'
+                  });
+                  return;
+                }
+
+                if (entry.attributes.METHOD === 'SAMPLE-AES-CENC') {
+                  this.trigger('warn', {
+                    message: 'SAMPLE-AES-CENC is deprecated, please use SAMPLE-AES-CTR instead'
+                  });
+                }
+
+                if (entry.attributes.URI.substring(0, 23) !== 'data:text/plain;base64,') {
+                  this.trigger('warn', {
+                    message: 'invalid key URI provided for Widevine'
+                  });
+                  return;
+                }
+
+                if (!(entry.attributes.KEYID && entry.attributes.KEYID.substring(0, 2) === '0x')) {
+                  this.trigger('warn', {
+                    message: 'invalid key ID provided for Widevine'
+                  });
+                  return;
+                }
+
+                // if Widevine key attributes are valid, store them as `contentProtection`
+                // on the manifest to emulate Widevine tag structure in a DASH mpd
+                this.manifest.contentProtection = {
```

Comment:
should we call this out specifically as `widevine` since it isn't standard hls?

Reply:
I guess the mpd parser calls it this as well?

Reply:
You mean renaming `contentProtection ` to `widevine`? We could, but that would require an additional change in VHS (and perhaps elsewhere), which already expects `contentProtection`. Calling it `contentProtection` maintains greater parity with mpd-parser.